File: xkcd/src/main/java/xyz/jienan/xkcd/model/WhatIfArticle.java
package xyz.jienan.xkcd.model;

import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;

@Entity
public class WhatIfArticle {

    @Id(assignable = true)
    public long num;

    public String title;

    public String featureImg;

    public String content;

    public String date;

    public boolean isFavorite = false;

    public boolean hasThumbed = false;

    public long thumbCount;

    public boolean hasRead = false;
}

File: xkcd/src/main/java/xyz/jienan/xkcd/model/WhatIfArticle.kt
package xyz.jienan.xkcd.model

import com.google.gson.annotations.SerializedName
import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id

@Entity
data class WhatIfArticle constructor(
        @Id(assignable = true)
        var num: Long = 0,
        var title: String = "",
        var featureImg: String? = null,
        var content: String? = null,
        @Transient
        var date: Long = 0L,
        @SerializedName("date")
        var dateInString: String = "",
        var isFavorite: Boolean = false,
        var hasThumbed: Boolean = false,
        var thumbCount: Long = 0
)

